pkg/agent/datamodel/sig_config.go

```diff
@@ -12,6 +16,151 @@ const (
+	return getCachedVersionsFromComponentsJSON(path.Join(path.Dir(filename), componentsFilePath))
+}
+
+func processManifest(manifest Manifest) {
+	CachedFromManifest["kubernetes"] = ProcessedManifest{
+		Versions: manifest.Kubernetes.Versions,
+	}
+	CachedFromManifest["runc"] = ProcessedManifest{
+		Versions:  manifest.Runc.Versions,
+		Pinned:    manifest.Runc.Pinned,
+		Installed: manifest.Runc.Installed,
+	}
+	CachedFromManifest["containerd"] = ProcessedManifest{
+		Versions: manifest.Containerd.Versions,
+		Pinned:   manifest.Containerd.Pinned,
+		Edge:     manifest.Containerd.Edge,
+	}
+	CachedFromManifest["nvidia-container-runtime"] = ProcessedManifest{
+		Versions: manifest.NvidiaContainerRuntime.Versions,
+	}
+	CachedFromManifest["nvidia-drivers"] = ProcessedManifest{
+		Versions: manifest.NvidiaDrivers.Versions,
+	}
+}
+
+func processComponents(components Components) {
+	for _, image := range components.ContainerImages {
+		componentName, err := processContainerImageTag(image.DownloadURL)
+		if err != nil {
+			panic(err)
+		}
+		processed := ContainerImage{
+			MultiArchVersions: image.MultiArchVersions,
+			Amd64OnlyVersions: image.Amd64OnlyVersions,
+		}
+		if len(image.PrefetchOptimizations) > 0 {
+			processed.PrefetchOptimizations = PrefetchOptimizations{
+				Version:  image.PrefetchOptimizations[0].Version,
+				Binaries: image.PrefetchOptimizations[0].Binaries,
+			}
+		}
+		CachedFromComponents[componentName] = processed
+	}
+}
+
+func getCachedVersionsFromManifestJSON(manifestFilePath string) Manifest {
+	data, err := os.ReadFile(manifestFilePath)
+	if err != nil {
+		panic(err)
+	}
+	data = bytes.ReplaceAll(data, []byte("#EOF"), []byte(""))
+	var manifest Manifest
+	if err = json.Unmarshal(data, &manifest); err != nil {
+		panic(err)
+	}
+	return manifest
+}
+
+func getCachedVersionsFromComponentsJSON(componentsFilePath string) Components {
+	data, err := os.ReadFile(componentsFilePath)
+	if err != nil {
+		panic(err)
+	}
+	var components Components
+	if err = json.Unmarshal(data, &components); err != nil {
+		panic(err)
+	}
+	return components
+}
+
+type Manifest struct {
+	Containerd struct {
+		Edge     string            `json:"edge"`
+		Versions []string          `json:"versions"`
+		Pinned   map[string]string `json:"pinned"`
+	} `json:"containerd"`
+	Runc struct {
+		Versions  []string          `json:"versions"`
+		Pinned    map[string]string `json:"pinned"`
+		Installed map[string]string `json:"installed"`
+	} `json:"runc"`
+	NvidiaContainerRuntime struct {
+		Versions []string `json:"versions"`
+	} `json:"nvidia-container-runtime"`
+	NvidiaDrivers struct {
+		Versions []string `json:"versions"`
+	} `json:"nvidia-drivers"`
+	Kubernetes struct {
+		Versions []string `json:"versions"`
+	} `json:"kubernetes"`
+}
+
+type ProcessedManifest struct {
+	Versions  []string
+	Pinned    map[string]string
+	Edge      string
+	Installed map[string]string
+}
+
+type Components struct {
+	ContainerImages []struct {
```

Comment:
since we won't do raw JSON we can just do something like this instead:
```
type Components struct {
    ContainerImages []ContainerImage `json:"containerImages"`
}

type ContainerImage struct {
	MultiArchVersions     []string `json:"multiArchVersions"`
	Amd64OnlyVersions     []string `json:"amd64OnlyVersions"`
	PrefetchOptimizations []PrefetchOptimization `json:"prefetchOptimizations"`
}

type PrefetchOptimization struct {
	Version  string `json:"version"`
	Binaries []string `json:"binaries"`
}
```

Reply:
btw, do we also need to include these? https://github.com/Azure/AgentBaker/blob/ddad5fa23550576d200628a1b21bae6ab7d689a9/vhdbuilder/packer/components.json#L368C1-L408C4